docs/fobnail-api.md

```diff
@@ -0,0 +1,459 @@
+on the evidence, Fobnail Token decides whether the platform is trustworthy or
+not. If Fobnail Token decides that the platform is trustworthy, **2.04**
+response is returned, **4.03** otherwise.
+
+If attestation is successful, Fobnail Token unlocks access to the Fobnail Token
+Services.
+
+### Fobnail Token Services
+
+Fobnail Token Services are available after successful platform attestation.
+Fobnail Token can store cryptographic keys (or other stuff) in its internal
+flash. Depending on the key's type (symmetric or asymmetric) and usage
+permissions, various operations are available. For symmetric keys, it is
+encryption and decryption. For asymmetric keys, it is signing, decryption, KDF,
+and public key read (The client extracts the public key and does encryption on
+its own).
+
+Key, once created, may not be read (except for the public key). Fobnail Token
+performs cryptographic operations on behalf of the client, which may be slow. If
+this behavior is undesired, Secure Storage may be used instead to store the key
+as a file inside Fobnail Token.
+
+| Endpoint Name        | Method | Arguments                        |
+| -------------------- | ------ | -------------------------------- |
+| /crypto              | POST   | Key name, type, etc.             |
+| /crypto/{id}/type    | GET    | None                             |
+| /crypto/{id}/encrypt | POST   | Data to encrypt                  |
+| /crypto/{id}/decrypt | POST   | Data to decrypt                  |
+| /crypto/{id}/sign    | PUT    | Data to sign                     |
+| /crypto/{id}/pub     | GET    | None                             |
+| /storage/fs/{name}   | GET    | File name                        |
+| /storage/fs/{name}   | PUT    | File name and file contents      |
+| /storage/fs/{name}   | DELETE | File name                        |
+
+If the platform has not been attested (or failed attestation), these APIs will
+return a **4.04** error. If attestation is successful, Fobnail Token checks
+whether a specified key exists and whether the platform has access to that key.
+If the key or file does not exist or there is no access **4.04** error is
+returned.
+
+> Note: currently, crypto API is not fully defined. Future versions of this
+> document will address this problem.
+
+#### GET /storage/fs/{name}
+
+The client sends a request to this endpoint to read a file. The file path is
+encoded as part of the URI. If the file exists (and if it's accessible),
+**2.05** response is sent with the file contents as the payload. `Max-Age`
+option **must** be set to 0 to prevent caching. ETag option **must not** be
+present in the response, and if present in the request, it **must** be ignored.
+
+> The GET method always returns the full contents of the file. Further versions
+> of this document may define API using the FETCH method from RFC8132.
+
+Fobnail Token attempts to open (or create) the requested file and returns
```

Comment:
Why create?

Reply:
Forgot to change that paragraph; previous version required to open the file before reading but we don't need to keep any context here as we do for some crypto operations so I simplified API.

Reply:
Fixed